scripts/core/editor3/components/Editor3.jsx

```diff
@@ -20,146 +21,137 @@ import Toolbar from './toolbar';
  * @description Editor3 is a draft.js based editor that support customizable
  *  formatting (shortly more features will come ...)
  */
-export class Editor3 extends React.Component {
-    constructor(props) {
-        super(props);
 
-        const initialContentState = stateFromHTML(props.value);
+export const getInitialState = (spellcheckerService, value, onChangeValue, readOnly, showToolbar, editorFormat) => {
```

Comment:
Why do you need this function? You can just place the same object returned here as the second argument to `createStore` in `Editor3Controller`:
```js
let store = createStore(editorReducers, {
     // ...
});
```
Alternatively, I would move this whole logic (`let store = ...`) to a `store` folder where you can do initialisations. Take a look at [this example](https://github.com/reactjs/redux/tree/master/examples/real-world/src) in the Redux repo.